### PR TITLE
Sprint 25 TLT-1705 Mailgun auth decorator

### PR DESCRIPTION
**Pull request by douglashall**

### CODE REVIEW

File: mailgun/decorators.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+from functools import wraps
+
+from django.conf import settings
+from django.utils.decorators import available_attrs
+from django.shortcuts import redirect
+from django.core.urlresolvers import reverse_lazy
+
+
+def authenticate(redirect_url=reverse_lazy('mailgun:auth_error'), raise_exception=False):
```
Comment (rascalking) [inline mailgun/decorators.py]: If it's not looked at, probably worth removing `raise_exception`.

File: mailgun/decorators.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def decorator(view_func):
+        @wraps(view_func, assigned=available_attrs(view_func))
+        def _wrapped_view(request, *args, **kwargs):
+            timestamp = request.POST.get('timestamp')
```
Comment (rascalking) [inline mailgun/decorators.py]: Not sure what the spec for the signature is, but typically you want to refuse to verify if the timestamp is outside a certain window.

File: mailgun/decorators.py (hunk 1 above)
Comment (douglashall) [inline mailgun/decorators.py]: Yes.

File: mailgun/views.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+from django.http import HttpResponse
+from django.views.decorators.http import require_http_methods
+
+
+logger = logging.getLogger(__name__)
+
+
+@require_http_methods(['GET'])
+def auth_error(request):
+    return HttpResponse(
```
Comment (rascalking) [inline mailgun/views.py]: Any reason not to use `[JsonResponse](https://docs.djangoproject.com/en/1.8/ref/request-response/#jsonresponse-objects)`?

File: mailgun/decorators.py (hunk 2 above)
Comment (douglashall) [inline mailgun/decorators.py]: Cool. Should be looking better now.

File: mailgun/views.py (hunk 3 above)
Comment (douglashall) [inline mailgun/views.py]: Nope. Changed.

**Comment by rascalking** [comment]
:+1:

**Pull request merged**